tests/TestHarness/Node.py

```diff
@@ -273,6 +273,58 @@ def isProducer():
+    def missedNextProductionRound(self):
+        # Cannot use producer_plugin's paused() endpoint as it does not
+        # include paused due to max-reversible-blocks exceeded.
+        # The idea is to find the scheduled start block of node's producer's
+        # next round. If that block is not produced, it means block production
+        # on the node is paused.
+
+        assert self.isProducer, 'missedNextProductionRound can be only called on a producer'
+
+        blocksPerProducer = 12
+
+        scheduled_producers = []
+        schedule = self.processUrllibRequest("chain", "get_producer_schedule")
+        for prod in schedule["payload"]["active"]["producers"]:
+            scheduled_producers.append(prod["producer_name"])
+        if Utils.Debug: Utils.Print(f'scheduled_producers {scheduled_producers}')
+
+        self.getInfo()
+        currBlockNum=self.lastRetrievedHeadBlockNum
+        currProducer=self.lastRetrievedHeadBlockProducer
+        blocksRemainedInCurrRound = blocksPerProducer - currBlockNum % blocksPerProducer - 1
+        if Utils.Debug: Utils.Print(f'currBlockNum {currBlockNum}, currProducer {currProducer}, blocksRemainedInCurrRound {blocksRemainedInCurrRound}')
+
+        # find the positions of currProducerPos and nodeProducer in the schedule
+        currProducerPos=0
+        nodeProducerPos=0
+        for i in range(0, len(scheduled_producers)):
+            if scheduled_producers[i] == currProducer:
+                currProducerPos=i
+            if scheduled_producers[i] == self.producerName:
+                nodeProducerPos=i
+
+        # find the number of the blocks to node producer's next scheduled round
+        blocksToNextScheduledRound = 0
+        if currProducerPos < nodeProducerPos:
+            # nodeProducerPos - currProducerPos - 1 is the number of producers
+            # from current producer to the node producer in the schedule
+            blocksToNextScheduledRound = (nodeProducerPos - currProducerPos - 1) * blocksPerProducer + blocksRemainedInCurrRound + 1
+        else:
+            # nodeProducerPos is the number of producers before node producer in the schedule
+            # len(scheduled_producers) - currProducerPos - 1 is the number
+            # of producers after node producer in the schedule
+            blocksToNextScheduledRound = (nodeProducerPos + (len(scheduled_producers)  - currProducerPos - 1)) * blocksPerProducer + blocksRemainedInCurrRound + 1
+
+        # find the block number of the node producer's next scheduled round
+        nextScheduledRoundBlockNum=currBlockNum + blocksToNextScheduledRound
+        timeout=blocksToNextScheduledRound/2 + 2 # leave 2 seconds for avoid flakiness
+        if Utils.Debug: Utils.Print(f'blocksToNextScheduledRound {blocksToNextScheduledRound}, nextScheduledRoundBlockNum {nextScheduledRoundBlockNum}, timeout {timeout}')
+
+        return not self.waitForBlock(nextScheduledRoundBlockNum, timeout=timeout)
```

Comment:
Even if the node doesn't produce during its round, wouldn't that same block number be produced by the next producer?

Reply:
Does the `self.waitForBlock` only waits for a block produced on the current node?

Reply:
No, any block produced or received

Reply:
So I wonder if the function `def missedNextProductionRound(self):` is correct?
It looks like it assumes that all producers will produce all the blocks of their rounds except for self.
For example if the other producer was paused, we would still time out and (incorrectly) believe this mode didn't produce during its round.

Reply:
Yes, seems like this could just use `waitForProducer()` with a timeout instead.